### DISCUSSION TITLE
How to get values from yaml in different profiles (separated with '---')

### DISCUSSION THREAD

**Discussion by haimiandbb** [mikefarah/yq#1353] [Q&A]
YAML Demo manager.yaml

spec:
  serviceName: "cache-manager-0"
  selector:
    matchLabels:
      app: cache-manager-0
  replicas: 1
  template:
    spec:
      containers:
        - name: manager
          image: cache-manager:master-202208120524
 \---
 spec:
  serviceName: "cache-manager-1"
  selector:
    matchLabels:
      app: cache-manager-1
  replicas: 1
  template:
    spec:
      containers:
        - name: manager
          image: cache-manager:master-202208120404

I use command in groovy: def res = sh returnStdout: true, script: "yq r manager.yaml 'spec.template.spec.containers.(name==manager).image' "
only get the first (res = cache-manager:master-202208120524)
What should I do to get the second one?
And how to update the second one ?

**Comment by mikefarah** [comment]
- You are using a old, unsupported version of yq - please update to the latest one.
- Please wrap the sample yaml in a code block (triple backticks) - it's very hard to read and understand the structure.

**Reply by mikefarah** [reply]
Assuming that v3 of yq - the user docs are still up here: https://mikefarah.gitbook.io/yq/v/v3.x/commands/read#read-from-all-documents
